roofit/hs3/src/RooJSONFactoryWSTool.cxx

```diff
@@ -742,25 +745,35 @@ JSONNode *RooJSONFactoryWSTool::exportObject(const RooAbsArg *func)
    elem["type"] << dict->second.type;
 
    size_t nprox = func->numProxies();
+   
    for (size_t i = 0; i < nprox; ++i) {
       RooAbsProxy *p = func->getProxy(i);
 
+      // some proxies start with a "!". This is a magic symbol that we don't want to stream
       std::string pname(p->name());
       if (pname[0] == '!')
          pname.erase(0, 1);
 
       auto k = dict->second.proxies.find(pname);
       if (k == dict->second.proxies.end()) {
          std::cerr << "failed to find key matching proxy '" << pname << "' for type '" << dict->second.type
-                   << "', skipping" << std::endl;
+                   << "', encountered in '" << func->GetName() << "', skipping" << std::endl;
          return nullptr;
       }
 
+      // empty string is interpreted as an instruction to ignore this value
+      if(k->second.size() == 0) continue;
+
       if (auto l = dynamic_cast<RooListProxy *>(p)) {
          elem[k->second].fill_seq(*l, [](auto const &e) { return e->GetName(); });
       }
       if (auto r = dynamic_cast<RooRealProxy *>(p)) {
-         elem[k->second] << r->arg().GetName();
+	// RooConstVars are just defined by their value, so we don't need to turn them into strings
```

Comment:
That is not true. Some RooConstVars are indeed just representing their value, and even have a name that is equivalent to the value. But this is only a subset of them (the ones created via `RooFit::RooConst()`)

I'll try to find another solution to write the values out as numbers, if this is the kind of RooConstVar where the name is equal to the value